lib/AppInfo/Application.php

```diff
@@ -132,4 +149,66 @@ public function register(IRegistrationContext $context): void {
 		$context->registerSearchProvider(SearchProvider::class);
 		$context->registerDashboardWidget(PollWidget::class);
 	}
+
+	/**
+	 * Register some Services
+	 */
+	private function registerServices(IRegistrationContext $context) {
+		$context->registerService(UserMapper::class, function (ContainerInterface $c): UserMapper {
+			return new UserMapper(
+				$c->get(IDBConnection::class),
+				$c->get(ISession::class),
+				$c->get(IUserSession::class),
+				$c->get(IUserManager::class),
+				$c->get(LoggerInterface::class),
+			);
+		});
```

Comment:
Is all this needed? This is the default DI behavior to my knowledge.

Reply:
I was annoyed by the unit tests, where every deopendency needed to be instatiated. 
If there is a DI possibility for the tests, this is indeed unnecessary.

Reply:
Usually you use Mocks for the tests

Reply:
Yeah. Tests. See https://github.com/nextcloud/polls/issues/3147 😉